File: CSharp/Patterns/Strategy/Ducks/Duck.cs
using System;
using Patterns.Strategy;
using Patterns.Strategy.Behaivours;

namespace Patterns.Strategy.Ducks
{
	public abstract class Duck
	{
		IFlyBehaviour flybehaviour;
		IQuackBehaviour quackbehavior;
		
		public Duck ()
		{
		}
		
		public void setFlyBehaivour(IFlyBehaviour behaviour)
		{
			flybehaviour = behaviour;
		}
		
		public void setQuackBehaivour(IQuackBehaviour behaivour)
		{
			quackbehavior = behaivour;
		}
		
		public void performFly()
		{
			flybehaviour.fly();
		}
		
		public void performQuack()
		{
			quackbehavior.quack();
		}
	}
}

File: PHP/Strategy/Ducks/Duck.php
<?php
 /**
 * Duck.php
 * 
 * An abstract Duck
 * @author Indika Piyasena <indika@webmolecule.co.za>
 * @version 1.0
 */
 
abstract class Duck {

	protected $flyBehaviour;

	public function setFlyBehaviour($flyBehaviour)
	{
		$this->flyBehaviour = $flyBehaviour;
	}

	public function performFly()
	{
		$this->flyBehaviour->fly();
	}


}
